lib/ansible/modules/systemd_service.py

```diff
@@ -484,17 +484,17 @@ def main():
             masked = out.strip() == "masked"
 
             if masked != module.params['masked']:
-                result['changed'] = True
-                if module.params['masked']:
-                    action = 'mask'
-                else:
-                    action = 'unmask'
+                action = "mask" if module.params['masked'] else "unmask"
 
                 if not module.check_mode:
                     (rc, out, err) = module.run_command("%s %s '%s'" % (systemctl, action, unit))
                     if rc != 0:
                         # some versions of system CAN mask/unmask non existing services, we only fail on missing if they don't
                         fail_if_missing(module, found, unit, msg='host')
+                        if "Failed to mask" in err:
```

Comment:
would add comment:
# here if service was not missing, but failed for other reasons

I would not even make it conditional